src/vm/analysis/type_checker/natives/mod.rs

```diff
@@ -300,27 +300,27 @@ impl TypedNativeFunction {
             Hash160 =>
                 Simple(SimpleNativeFunction(FunctionType::UnionArgs(
                     vec![TypeSignature::max_buffer(),
-                         TypeSignature::IntType],
+                         TypeSignature::UIntType],
                     BUFF_20.clone()))),
             Sha256 =>
                 Simple(SimpleNativeFunction(FunctionType::UnionArgs(
                     vec![TypeSignature::max_buffer(),
-                         TypeSignature::IntType],
+                         TypeSignature::UIntType],
```

Comment:
The hash functions should probably be unions of ints, uints or buffs, rather than just uints or buffs.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1129/commits/8489cb6e963bd0c6cd6b8fcd598c0abff5cbdd8f